File: 2022/4a.py
with open("inputs/day4.txt") as f:
    assignments = f.read()

count = 0
for assignment in assignments.split("\n"):
    as1, as2 = assignment.split(",")
    as1 = list(map(int, as1.split("-")))
    as2 = list(map(int, as2.split("-")))

    if as1[0] <= as2[0] <= as2[1] <= as1[1]:
        count += 1
    elif as2[0] <= as1[0] <= as1[1] <= as2[1]:
        count += 1
print(count)

File: 2022/rust/src/bin/4a.rs
fn main() {
    let test = "2-4,6-8
2-3,4-5
5-7,7-9
2-8,3-7
6-6,4-6
2-6,4-8";
    let input_str = include_str!("../../inputs/day4.txt");
    let mut count = 0;
    for line in input_str.lines() {
        let mut ranges = line
            .split(|c| c == '-' || c == ',')
            .map(|num| num.parse::<u8>().unwrap());
        let (a1, a2, b1, b2) = (ranges.next(), ranges.next(), ranges.next(), ranges.next());
        let (a_range, b_range) = (a1..=a2, b1..=b2);
        if a_range.contains(&b1) && a_range.contains(&b2) {
            count += 1;
        }
        if b_range.contains(&a1) && b_range.contains(&a2) {
            count += 1;
        }
    }
    println!("{:?}", count);
}
